stacks-signer/src/runloop.rs

```diff
@@ -164,45 +176,211 @@ impl<C: Coordinator> RunLoop<C> {
         }
     }
 
-    /// Process the event as both a signer and a coordinator
-    fn process_event(
+    /// Handle the block validate response returned from our prior calls to submit a block for validation
+    fn handle_block_validate_response(&mut self, block_validate_response: BlockValidateResponse) {
+        match block_validate_response {
```

Comment:
Should the signer binary track block validation responses directly? This seems like state that will be important for the signer binary to have: that way if a sign request comes in, the signer knows whether or not it has already validated (or rejected) the requested proposal.

Reply:
We do have a ticket in plcae to add state to the signers. I do believe we will need it for not only that but to know what transactions we expect to see in a block. I think this should remain a second ticket though considering the size of this change/because I believe we could work on it asyncronously at the same time as the serialization of the block ticket that Joey is assigned to.

Reply:
I have made two tickets that track the need for local state for both block requests/responses/sign requests and the need to track transactions to verify a block against

https://github.com/stacks-network/stacks-core/issues/4242
https://github.com/stacks-network/stacks-core/issues/4243

Reply:
Sounds good to me